File: code/801二进制中1的个数.cpp
#include <iostream>
#include <cstring>
#include <algorithm>
using namespace std;

int n, x;

int lowbit(int x) {
	return x & -x;
}

int main() {
	cin >> n;
	while (n--) {
		cin >> x;
		int res = 0;
		while (x) {
			x -= lowbit(x);
			res++;
		}
		cout << res << " ";
	}

	return 0;
}

File: code/0801/801二进制中1的个数.py
def lowbit(x):
    return x & -x


if __name__ == "__main__":
    n = int(input())
    g = list(map(int, input().split()))
    for x in g:
        res = 0
        while x:
            x -= lowbit(x)
            res += 1
        print(res, end=' ')
